### PR TITLE
Initial commit for Pure Storage FlashBlade module

### PR DESCRIPTION
**Pull request by sdodsley**
##### SUMMARY
Storage module for Pure Storage FlashBlade device. Module for create/delete/modify filesystem,

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
purefb_fs

##### ANSIBLE VERSION
2.6

##### ADDITIONAL INFORMATION
Add support FlashBlade modules into core pure module and add new doc fragment for the FlashBlade modules.
Create common document_fragment for all Pure Storage modules.

Update Pure Storage FlashArray module to point to merged document fragment.

### CODE REVIEW

File: lib/ansible/modules/storage/purestorage/purefb_fs.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    description:
+      - Define whether to NFS protocol is enabled for the filesystem.
+    required: false
+    type: bool
+    default: true
+  nfs_rules:
+    description:
+      - Define the NFS rules in operation.
+    required: false
+    type: bool
```
Comment (gundalow) [inline lib/ansible/modules/storage/purestorage/purefb_fs.py]: I think this line needs removing

File: lib/ansible/modules/storage/purestorage/purefb_fs.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    default: '*(rw,no_root_squash)'
+  smb:
+    description:
+      - Define whether to SMB protocol is enabled for the filesystem.
+      - Define whether to SMB protocol is enabled for the filesystem.
```
Comment (gundalow) [inline lib/ansible/modules/storage/purestorage/purefb_fs.py]: Remove duplicated line

File: lib/ansible/modules/storage/purestorage/purefb_fs.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    required: false
+    type: bool
+    default: false
+  http:
+    description:
+      - Define whether to HTTP/HTTPS protocol is enabled for the filesystem.
```
Comment (gundalow) [inline lib/ansible/modules/storage/purestorage/purefb_fs.py:65]: Doesn't sound correct.
`Define whether HTTP/HTTPS protocol should be enabled for the filesystem.`?

Comment (sdodsley) [reply lib/ansible/modules/storage/purestorage/purefb_fs.py:65]: @gundalow The device I'm doing this module for can have HTTP or HTTPS access to the filesystem as well as classic NFS/SMB. Newer applications can use these protocols to stream files

File: lib/ansible/modules/storage/purestorage/purefb_fs.py [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+try:
+    from purity_fb import FileSystem, ProtocolRule, NfsRule
+except ImportError:
+    HAS_PURITY_FB = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pure import get_blade, purefb_argument_spec
+
+
+def human_to_bytes(size):
```
Comment (sivel) [inline lib/ansible/modules/storage/purestorage/purefb_fs.py]: Is it not possible to use `ansible.module_utils.basic.human_to_bytes` here?

File: lib/ansible/utils/module_docs_fragments/purestorage.py [hunk 5]
```diff
@@ -21,6 +21,34 @@ class ModuleDocFragment(object):
...
     DOCUMENTATION = '''
+options:
+  - See seperate platform section for more details
+requirements:
+  - See seperate platform section for more details
+notes:
+  - Ansible modules are available for the following Pure Storage products: FlashArray, FlashBlade
+'''
+
+    # Documentation fragment for FlashBlade
```
Comment (sivel) [inline lib/ansible/utils/module_docs_fragments/purestorage.py:32]: You have defined these in capital letters here, but used them as lowercase in the actual modules.  This wont work.

Comment (sdodsley) [reply lib/ansible/utils/module_docs_fragments/purestorage.py:32]: @sivel I took examples of this from other files and they seem to work by using capitals in the fragment file and lowercase in the actual code. Is the best practice to use all lower or all upper case in both?

Comment (sivel) [reply lib/ansible/utils/module_docs_fragments/purestorage.py:32]: Looks like I am wrong, seems to be standard to do it this way.

**Comment by jbilbro** [comment]
I have tested this purefb_fs module on our FlashBlade and it works perfectly.  Please get this merged into 2.5!

Thanks, @sdodsley !

**Comment by gundalow** [comment]
@jbilbro Thank you for testing this

The cut off for new modules was 22 January 2018
https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_5.rst

So this will be for 2.6

Thank you for testing. You can use the module locally but put it in you `./library/` directory.

**Comment by gundalow** [comment]
rebuild_merge

**Pull request merged**